internal/controllers/database/sync.go

```diff
@@ -272,11 +272,41 @@ func (r *Reconciler) handleResourcesSync(
 	return Continue, ctrl.Result{Requeue: false}, nil
 }
 
+func (r *Reconciler) processSkipInitPipeline(
+	ctx context.Context,
+	database *resources.DatabaseBuilder,
+) (bool, ctrl.Result, error) {
+	r.Log.Info("running step processSkipInitPipeline")
+	r.Log.Info("Database initialization disabled (with annotation), proceed with caution")
+
+	r.Recorder.Event(
+		database,
+		corev1.EventTypeWarning,
+		"SkippingInit",
+		"Skipping initialization due to skip annotation present, be careful!",
+	)
+
+	return r.setInitDatabaseCompleted(
+		ctx,
+		database,
+		"Database initialization not performed because initialization is skipped",
```

Comment:
Minor, but I would change the wording slightly:
```suggestion
		"Database creation not performed because initialization is skipped",
```